python/lsst/summit/utils/astrometry/fastStarTrackerAnalysis.py

```diff
@@ -80,6 +107,214 @@ def tifToExp(filename):
     return exp
 
 
+def fitsToExp(filename):
+    """Open a fits file as an exposure.
+
+    Parameters
+    ----------
+    filename : `str`
+        The full path to the file to load.
+
+    Returns
+    -------
+    exp : `lsst.afw.image.Exposure`
+        The exposure.
+    """
+    exp = afwImage.ExposureF(filename)
+    return exp
+
+
+def openFile(filename):
+    """Open a file as an exposure.
+
+    Parameters
+    ----------
+    filename : `str`
+        The full path to the file to load.
+
+    Returns
+    -------
+    exp : `lsst.afw.image.Exposure`
+        The exposure.
+    """
+    if filename.endswith(".tif"):
+        return tifToExp(filename)
+    elif filename.endswith(".fits"):
+        return fitsToExp(filename)
+    else:
+        raise ValueError("File type not recognized")
+
+
+def dayObsToDateTime(dayObs):
+    """Convert a dayObs to a datetime.
+
+    Parameters
+    ----------
+    dayObs : `int`
+        The dayObs.
+
+    Returns
+    -------
+    datetime : `datetime`
+        The datetime.
+    """
+    return datetime.datetime.strptime(dayObsIntToString(dayObs), "%Y-%m-%d")
+
+
+def isStreamingModeFile(filename):
+    """Check if a filename is a streaming mode file.
+
+    Parameters
+    ----------
+    filename : `str`
+        The filename.
+
+    Returns
+    -------
+    isStreaming : `bool`
+        Whether the file is a streaming mode file.
+    """
+    # non-streaming filenames are like GC103_O_20240304_000009.fits
+    # streaming filenames are like GC103_O_20240304_000007_0001316.fits
+    # which is <camNum>_O_<dayObs>_<seqNum>_<streamSeqNum>.fits
+    # so 5 sections means streaming, 4 means normal
+    return len(os.path.basename(filename).split("_")) == 5
```

Comment:
You can also do `os.path.basename(filename).count("_") == 5`, FYI.

Reply:
Nice, thanks, I do prefer that actually. Note the fence-post error there though - needs to be `os.path.basename(filename).count("_") == 4` for the same behaviour!